src/sats_management.c

```diff
@@ -142,11 +145,14 @@ static void set_reference_sat(const u8 ref_prn, sats_management_t *sats_manageme
           j++;
         }
       }
-      else if (i == sats_management->num_sats-1) {
-        sats_management->prns[j] = old_ref;
-      }
     }
+    if (set_old_yet == 0) {
+      or_count++;
+      sats_management->prns[j] = old_ref;
+    }
+    assert (or_count == 1);
```

Comment:
`or_count` is just a check that we insert the old ref only once right? If its no longer useful then take it out otherwise if you think its a good cross-check then it can stay. I don't mind.